Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -91,17 +78,6 @@ static SessionTests()
                 OnAfterReset = (beforeFrozen, after) => { },
                 ManualResetFallback = (clientResetException) => { },
             };
-
-            // Just to check that we don't break previous code. Remove in next major version
-            var preserveObsoleteDiscardHandler = new DiscardLocalResetHandler
-            {
-                OnBeforeReset = (beforeFrozen) => { },
-                OnAfterReset = (beforeFrozen, after) => { },
-                ManualResetFallback = (clientResetException) => { },
-            };
```

Comment:
Technically, it's needed, but the session tests were taking a whole lot of time, the majority of which was spent testing obsolete behavior. Since we're doing a major version release soon and we're not expecting to do any SDK code changes related to the client reset handlers, I figured it would be a small risk to delete the obsolete tests earlier.

Reply:
I thought this was the reasoning behind and I think it's a good idea, but then I saw that you left most of the other obsolete tests so I got confused and I asked.
So, why removing only some and not all? What's the reasoning behind?

Reply:
Fair. We'll remove all these tests in 3 months or so anyway, so it's fine for now. I was just curious.

Reply:
A bunch of things have changed related to schema and rules on the server, though I'm not following exactly what 😄 I'm sure when those were added they were necessary, but don't seem to be anymore and were creating issues with asymmetric sync when the rule was created before the schema.

Reply:
I think this answer was for the question on [BaasClient](https://github.com/realm/realm-dotnet/pull/3096#discussion_r1071440203)

Reply:
Oh, yeah, wrong thread. That was indeed about the rules and why they're gone.